### PR TITLE
feat(models): add PTv3 segmentation decoder head

### PR DESCRIPTION
**Pull request by amadeuszsz**
## Summary

- Add `PTv3SegDecoderHead`: attention-based decoder (per-stage unpooling blocks + classifier) with cross-entropy + Lovasz loss, replacing the linear segmentation head; configured via `seg_head_base` / `seg_head_tiny`.
- Rename the PTv3 backbone to **encoder** everywhere: code, config keys (`model.encoder`, `freeze_encoder`), file names (`encoders/ptv3.py`, `encoder_base.yaml`, `encoder_tiny.yaml`), and the exported ONNX module (`encoder`).
- Rework the detection branch into `PTv3DetBEVNeck` (`bev_neck.yaml`): `PTv3DetFeatureFusion` fuses the deepest encoder stage into its parent stage (one non-destructive unpooling step), followed by BEV projection and the dilated BEV encoder. Detection-only and joint segdet models share one neck implementation and state-dict layout.
- Export the encoder with per-stage outputs (`point_feat_0` … `point_feat_4`); the detection head module consumes the two deepest stages, the segmentation head module emits `pred_labels` / `pred_probs`.
- Support multiple annotation files per split in the segdet datamodule (`AnnotationSource`): each source declares its supervision (`det3d` / `seg3d`) and a `repeat` factor; sources without detection supervision contribute no detection loss.
- Split trainable parameters into attention-block and default groups (`split_block_parameters`) to allow per-group learning rates.
- Collation cleanup: `strength` is folded into `feat` by `BuildPointFeatures`; `inverse` is collated only where point-level outputs exist (seg3d / multi).
- Remove `MergeObjects3D` from the nuScenes detection config: it referenced `merge_objects`, which is not defined for nuScenes; T4 configs keep it.

## Checklist

- [ ] You've checked our [contribution overview](https://tier4.github.io/autoware-ml/contributing/contribution-overview/).
- [ ] Your PR follows our [pull request guidelines](https://tier4.github.io/autoware-ml/contributing/pull-request-guidelines/).

### CODE REVIEW

File: autoware_ml/models/segmentation3d/ptv3_base.py [hunk 1]
```diff
@@ -3,91 +3,139 @@
...
                 x_max, y_max, z_max]`` used to derive sparse shape for export.
-            freeze_backbone: When ``True``, the backbone is permanently kept
+            freeze_encoder: When ``True``, the encoder is permanently kept
                 in eval mode with its parameters frozen.
             **kwargs: Keyword arguments forwarded to :class:`BaseModel` (and
                 further up the MRO chain).
         """
         super().__init__(**kwargs)
-        self.backbone = backbone
+        self.encoder = encoder
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/models/segmentation3d/ptv3_base.py:125]: Don't care about legacy support!

File: autoware_ml/configs/tasks/detection3d/ptv3/voxel005_51m_nuscenes.yaml [hunk 2]
```diff
@@ -135,9 +134,9 @@ model:
       rot,
       vel,
     ]
-  bev_projector:
-    bev_stride: 8
-    output_shape: [256, 256]
+  bev_neck:
+    bev_projector:
+      output_shape: [256, 256]
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/configs/tasks/detection3d/ptv3/voxel005_51m_nuscenes.yaml:139]: `bev_stride` not needed anymore https://github.com/tier4/autoware-ml/pull/68/commits/bf7cda8605058b6948abe738736f8ab403812fdb

File: autoware_ml/configs/tasks/multi/ptv3/voxel005_51m_nuscenes.yaml [hunk 3]
```diff
@@ -135,9 +137,9 @@ datamodule:
         output_key: feat
 
 model:
-  bev_projector:
-    output_shape: [256, 256]
-    bev_stride: 8
+  bev_neck:
+    bev_projector:
+      output_shape: [256, 256]
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/configs/tasks/multi/ptv3/voxel005_51m_nuscenes.yaml:142]: `bev_stride` not needed anymore https://github.com/tier4/autoware-ml/pull/68/commits/bf7cda8605058b6948abe738736f8ab403812fdb

File: autoware_ml/configs/tasks/multi/ptv3/voxel012_122m_t4dataset_j6gen2.yaml [hunk 4]
```diff
@@ -154,9 +156,9 @@ datamodule:
         output_key: feat
 
 model:
-  bev_projector:
-    output_shape: [256, 256]
-    bev_stride: 8
+  bev_neck:
+    bev_projector:
+      output_shape: [256, 256]
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/configs/tasks/multi/ptv3/voxel012_122m_t4dataset_j6gen2.yaml:161]: `bev_stride` not needed anymore https://github.com/tier4/autoware-ml/pull/68/commits/bf7cda8605058b6948abe738736f8ab403812fdb

File: autoware_ml/configs/tasks/detection3d/ptv3/voxel012_122m_t4dataset_j6gen2.yaml [hunk 5]
```diff
@@ -141,9 +143,9 @@ model:
       rot,
       vel,
     ]
-  bev_projector:
-    bev_stride: 8
-    output_shape: [256, 256]
+  bev_neck:
+    bev_projector:
+      output_shape: [256, 256]
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/configs/tasks/detection3d/ptv3/voxel012_122m_t4dataset_j6gen2.yaml:148]: `bev_stride` not needed anymore https://github.com/tier4/autoware-ml/pull/68/commits/bf7cda8605058b6948abe738736f8ab403812fdb
